auth4genai/mcp/intro/overview.mdx

```diff
@@ -0,0 +1,186 @@
+
+<Frame>
+  <img
+    className="hidden dark:block"
+    src="/img/mcp/mcp_diagram_dark.png"
+    alt="MCP Diagram"
+  />
+  <img
+    className="block dark:hidden"
+    src="/img/mcp/mcp_diagram_light.png"
+    alt="MCP Diagram"
+  />
+</Frame>
+
+Auth0's **Auth for MCP** lets developers securely and easily implement the authorization parts of the MCP spec with OAuth 2.1 and OpenID Connect. It provides sign in, standards based discovery and client registration, resource scoped tokens, and token exchange so you can control which agents connect, which resources they can access, and which actions they can perform.
+
+### User Authentication
+
+Require users to sign in before they or their AI agents connect to MCP servers. Auth0 Universal Login supports social, enterprise, and custom identity providers so users can sign in with existing credentials. Access tokens issued by Auth0 include identity details that MCP servers can validate on every request.
+
+For enterprise environments, **Workforce Identity for MCP** integrates your MCP deployment with your organization’s identity provider using **Single Sign-On (SSO)**. Auth0 connects to Okta, Entra ID, Ping, Google Workspace, and other IdPs so employees authenticate with existing credentials before any MCP interaction begins.
+
+<Card
+  title="Learn more about user authentication for MCP"
+  href="/mcp/intro/user-authentication"
+  icon="book"
+  horizontal
+  iconType="solid" />
+
+
+### Metadata Discovery and Client Registration for MCP
+
+Register MCP clients and servers using standards-aligned discovery and registration so agents connect only to approved resources. Auth0 validates client metadata, issues client credentials, and exposes the necessary discovery endpoints. This lets MCP clients find the right authorization server and request tokens without manual configuration.
+
+<Card
+  title="Learn how to register your MCP client application"
+  href="/mcp/guides/registering-your-mcp-client-application"
+  icon="book"
+  horizontal
+  iconType="solid" />
+
+### Call Your APIs on User’s Behalf
+
+When an **AI agent** connects to your **MCP server**, the server may need to call your internal APIs to read data or trigger workflows. Those APIs need tokens that reflect who the user is and what the server is allowed to do.
+
+**Auth0 Custom Token Exchange** enables that delegation. The MCP server exchanges the token it received from the client for a new, short-lived access token scoped to the internal API. This lets **MCP servers** call internal APIs on behalf of users while **Auth0** issues and governs access centrally.
+
+<Card
+  title="Learn more about how to use custom token exchange to call your APIs on user's behalf"
+  href="/mcp/intro/call-your-apis-on-users-behalf"
+  icon="book"
+  horizontal
+  iconType="solid" />
+
+### Call Third-Party APIs on User’s Behalf
```

Comment:
We don't currently have sub-sections about calling Third-Party APIs using Token Vault. Is this paragraph meant to be kept for later, or we are fine leaving it here with no concrete instructions/details?